make.go

```diff
@@ -54,10 +72,19 @@ func (g *castGen) String() string {
 
 func (g *castGen) value(t *T) any {
 	v := g.gen.value(t)
+	if v == nil {
+		return nil
+	}
```

Comment:
Why is this check here?

Reply:
I was seeing nil pointer exceptions in cases where `v` was nil. After double checking [`reflect.ValueOf`](https://pkg.go.dev/reflect#ValueOf) I suspect the issue stems from `reflect.Value.Convert` as `reflect.ValueOf(nil)` returns a zero `reflect.Value`. 

I'll add a comment and/or I could change this to:
```go
v := reflect.ValueOf(g.gen.value(t))
if v.IsNil() {
    return nil
}
return v.Convert(g.type).Interface()
```

Reply:
In what cases `.value()` is `nil`? It should not be, ever.

Reply:
Ah! I figured it out. I was attempting to skip over a pointer field but missed the `*` so I had overridden a struct generator with `rapid.Just[any](nil)`. I'll see if I can make the error message a bit more friendly and add a regression test.